sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -65,10 +65,10 @@ trait CheckpointFileManager {
   /** Open a file for reading, or throw exception if it does not exist. */
   def open(path: Path): FSDataInputStream
 
-  /** List the files in a path that match a filter. */
+  /** List the files/directories in a path that match a filter. */
   def list(path: Path, filter: PathFilter): Array[FileStatus]
```

Comment:
It turns out that some usage of `list` method break the assumption that it can only return files. So I change the method doc to make it explicitly defined.